tests/e2e/utils/use-request-interception.js

```diff
@@ -26,3 +26,45 @@ export function useRequestInterception( callback ) {
 		page.off( 'request', requestHandler );
 	};
 }
+
+// TODO: Remove this function and it's usages as part of #6433.
+/**
+ * Adds shared request handler for intercepting requests.
+ *
+ * Used intercept HTTP requests during tests with a custom handler function.
+ * Returns a function that, when called, stops further request
+ * handling/interception.
+ *
+ * @since n.e.x.t
+ *
+ * @param {Array.<{isMatch: Function, getResponse: Function}>} requestCases An array of request cases to add to the shared request interception.
+ * @return {Object} Methods that can be called to remove the added handler function from the page and add new request cases.
+ */
+export function useSharedRequestInterception( requestCases ) {
+	const cases = [ ...requestCases ];
+	const requestHandler = ( request ) => {
+		// Prevent errors for requests that happen after interception is disabled.
+		if ( ! request._allowInterception ) {
+			return;
+		}
+
+		const requestCase = cases.find( ( { isMatch } ) => {
+			return isMatch( request );
+		} );
+
+		if ( requestCase ) {
+			request.respond( requestCase.getResponse( request ) );
+		} else {
+			request.continue();
+		}
+	};
+
+	page.on( 'request', requestHandler );
+
+	return {
+		cleanUp: () => page.off( 'request', requestHandler ),
+		addRequestCases: ( newCases ) => {
+			cases.push( ...newCases );
+		},
```

Comment:
How about this is the only way to add cases? If this were the only way to add cases we could eliminate the possibility of typos in the "case schema" by using two params instead.

E.g.

```js

const sri = useSharedRequestInterception = function() {
	// ...
	return {
		cleanUp: () => // ...
		addMatchHandler = ( isMatch, getResponse ) => ( { isMatch, getResponse } ),
	}
};
sri.addMatchHandler( matcher, response );
sri.addMatchHandler( matcher, response );
sri.addMatchHandler( matcher, response );
```
Something like that, might simplify the use a bit. What do you think?

Reply:
Probably not worth it if this is only temporary (removing in #6433).

Reply:
Yes, if we are changing the `addMatchHandler` method, I would like to change the parameter of the `useSharedRequestInterception` function to maintain the same API. However, as you said, it's not worth considering it is a temporary solution.